stacks/dotnet31/devfile.yaml

```diff
@@ -0,0 +1,50 @@
+schemaVersion: 2.1.0
+metadata:
+  name: dotnet50
+  displayName: .NET 3.1
+  description: Stack with .NET 3.1
+  icon: https://developers.redhat.com/sites/default/files/styles/large/public/dotnet-logo3x3.jpg
+  projectType: dotnet
+  version: 1.0.0
+
+starterProjects:
+  - name: dotnetcore
+    git:
+      checkoutFrom:
+        remote: origin
+        revision: dotnetcore-3.1
+      remotes:
+        origin: https://github.com/redhat-developer/s2i-dotnetcore-ex
+    subDir: app
+
+components:
+- name: dotnet
+  container:
+    image: registry.access.redhat.com/ubi8/dotnet-31:3.1
+    mountSources: true
+    env:
+      - name: CONFIGURATION
+        value: Debug
+    endpoints:
+    - name: http-8080
+      targetPort: 8080
+
+commands:
+- id: build
+  exec:
+    workingDir: ${PROJECTS_ROOT}
+    commandLine: dotnet build -c $CONFIGURATION
+    component: dotnet
+    group:
+      isDefault: true
+      kind: build
+
+- id: run
+  exec:
+    workingDir: ${PROJECTS_ROOT}
+    commandLine: dotnet ./bin/$CONFIGURATION/netcoreapp3.1/app.dll
```

Comment:
`app.dll` is specific to the example application.
This should be a parameter too.
The default value can be derived from the folder containing the application (for the example it is `app`) and adding `.dll` to it.

Reply:
Yes, in our s2i image we also derive it from the project file name.
The project file can be for C# (`.csproj`), it may also be for F# (`.fsproj`) or VB (`vbproj`).

`STARTUP_PROJECT` can be set to a `.??proj` file, but the s2i builder also allows the user to specify its parent folder, and then looks for the project file in the folder. I don't know how smart you can make this `Devfile`.